backends/qualcomm/CMakeLists.txt

```diff
@@ -23,15 +23,66 @@ get_filename_component(
   _common_include_directories "${EXECUTORCH_SOURCE_DIR}/.." ABSOLUTE
 )
 
-if(NOT DEFINED QNN_SDK_ROOT)
-  message(
-    FATAL_ERROR
-      "Please define QNN_SDK_ROOT, e.g. cmake <..> -DQNN_SDK_ROOT=<...>"
-  )
-elseif(CMAKE_TOOLCHAIN_FILE MATCHES ".*(iOS|ios\.toolchain)\.cmake$")
+if(CMAKE_TOOLCHAIN_FILE MATCHES ".*(iOS|ios\.toolchain)\.cmake$")
   message(FATAL_ERROR "ios is not supported by Qualcomm AI Engine Direct")
 endif()
 
+set(_qnn_host_is_linux_x86 FALSE)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR MATCHES
+                                          "^(x86_64|amd64|i.86)$"
+)
+  set(_qnn_host_is_linux_x86 TRUE)
+endif()
+
+if(NOT DEFINED QNN_SDK_ROOT)
+  if(_qnn_host_is_linux_x86)
+    set(_qnn_default_sdk_dir "${CMAKE_CURRENT_BINARY_DIR}/sdk/qnn")
```

Comment:
I feel like it will cause weird logic like pip install will have the sdk in pip site packages but then when we clone the executorch repro for runtime, we are going to download a seperate version inside executorch github. I'm not sure what the behavior will be

Reply:
> when we clone the executorch repro for runtime, we are going to download a seperate version inside executorch github.

How is the behavior different than before this PR?

Reply:
It's true that we have a version check. But tracking down where the malformed sdk path is quite annoying especially we're using the generic `LD_LIBRARY_PATH`. If possible, I'd prefer to have one sdk path in the system